source/distributed/shared_tria.cc

```diff
@@ -76,14 +146,84 @@ namespace parallel
                   active_halo_layer.find(cell) == active_halo_layer.end())
                 cell->set_subdomain_id(numbers::artificial_subdomain_id);
             }
+
+          // loop over all cells in multigrid hierarchy and mark artificial:
+          if (settings & construct_multigrid_hierarchy)
+            {
+              std_cxx11::function<bool (const typename parallel::shared::Triangulation<dim,spacedim>::cell_iterator &)> predicate
```

Comment:
Oh, now I get it. Your point is that the flag ``construct_multigrid_hierarchy`` is redundant when using custom partitioning. While this is true, I have two arguments to keep things as is:
1. If I want to switch from metis to custom in a particular program, I would also need to fiddle with the settings.
2. I can envision a reusable "custom" routine that looks at the settings to decide if multigrid cells need to be partitioned or not.